scripts/cli.py

```diff
@@ -0,0 +1,170 @@
+"""The CLI tool to manipulate kernel tuner cache files.
+
+Works with the cache file manipulation library (cache.py),
+conversion functionality (convert.py) and other helper functions (defined in cli_functionality.py).
+
+Basic usage:
+$ poetry run ktcache {convert, t4, delete-line, get-line, merge}
+
+We can:
+   - `convert`: using the functionality from convert.py one can convert to a specified version.
+      Within conversion (all required yet nonpositional arguments):
+         - `-i/--infile`: the input file to read from.
+         - `-T/--target`: The version to convert to. 
+         - `-o/--output`: The converted output file.
+
+   - `t4`: using the functionality from convert.py one can convert to T4 format.
+      Within t4 conversion:
+         - `-i/--infile`: the input file to read from.
+         - `-o/--output`: the t4-converted output file.
+
+   - `delete-line`: using the cache library one can delete from a certain cachefile; only works with the latest 
+      schema_version. 
+      We have arguments:
+         - <infile>: the input file to delete the entry.
+         - `--key`: The key to try and delete.
+         - ``: the new output file.
+
+   - `get-line`: using the cache library one can get a certain cacheline entry from the input cachefile. 
+      Prints in JSON.
+      We have arugments:
+         - <infile>: the input file to get the cacheline entry from.
+         - (Required) `--key`: The key to check if present in the cachefile. 
+
+   - `merge`: merge several (>=2) cache files that are of the same jsonschema and have equivalent metadata, to the 
+      latest schema_version.
+      Arguments are (all required):
+         - <files>: The list of (space separated) input files to read in order to merge the cachefiles.
+         - `-o, --output`: The output file to write the merged result to.
+
+Example usages:
+$ cli convert --infile 1.json -T 1.1.0 --out 2.json
+$ cli delete-line 1.json --key <key> 
+$ cli get-line 1.json --key <key>
+$ cli merge 1.json 2.json 3.json 4.json -o merged.json
+$ cli t4 --in unconverted.json --out converted.json
+
+
+"""

```

Comment:
Deze documentatie is dubbelop. Er zijn zo dan drie plaatsen waar de documentatie bijgehouden moet worden, namelijk in `cli_functionality.py`, `cli.py`, en het commando `cli --help`.